src/test/java/pl/gacik/coordinates/SimpleCoordinates2DTests.java

```diff
@@ -0,0 +1,158 @@
+package pl.gacik.coordinates;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.testng.Assert;
+
+import java.util.Random;
+
+@Test(groups = {"SimpleCoordinates2DTests"})
+public class SimpleCoordinates2DTests {
+
+    @DataProvider(name = "10randomCoordinates")
+    public static Object[][] randomCoordinates() {
+        Random generator = new Random();
+        Object[][] result = new Object[10][2];
+        for(int i=0 ; i<10; i++) {
+            result[i][0] = generator.nextInt();
+            result[i][1] = generator.nextInt();
+        }
+        return result;
+    }
+
+    @Test(dataProvider = "10randomCoordinates")
+    public void objectCanBeInitializedCorrectlyTest(Integer x, Integer y) {
+        Coordinates2DInterface coordinates2D = new SimpleCoordinates2D(x, y);
+    }
+
+    @Test(dataProvider = "10randomCoordinates")
+    public void twoObjectsWithSameCoordinatesIsTheSameTest(Integer x, Integer y) {
+        // given
+        Coordinates2DInterface first = new SimpleCoordinates2D(x,y);
+        Coordinates2DInterface second = new SimpleCoordinates2D(x,y);
+        // when - then
+        Assert.assertEquals(first, second);
+    }
+
+    @Test(dataProvider = "10randomCoordinates")
+    public void getXReturnInitializedXValueCorrectlyTest(Integer x, Integer y) {
```

Comment:
I know from Tomek, that having 'Test' in test method name is redundant :P

Good option (but not always) for naming is starting from world 'should' do something.

In you test I would name it 'shouldInitXValueCorrectly'.